src/tweak/tree.zig

```diff
@@ -0,0 +1,154 @@
+const std = @import("std");
+
+pub fn MerkleTree(comptime TweakHash: type) type {
```

Comment:
better to use `https://github.com/blockblaz/ssz.zig` by defining the ssz type and doing hashtreeroot

similarly for merkle path, if merkle path/proof is not in ssz lib above then PR to add there